### PR TITLE
Add SVG support to NFT Gallery

### PR DESCRIPTION
**Pull request by michalszorad**
Fixes [YOMO-489](https://emurgo.atlassian.net/jira/software/c/projects/YOMO/issues/YOMO-489)

React native does not seem to support SVGs by default, which means I've opted out to use `react-native-svg` to display remote SVGs.

`react-native-svg` however does not support `blurRadius` making it impossible to blur out the SVG when its moderation status is `consent`. In order to fix the blur issue, I've tried to use `@react-native-community/blur` but this library crashes when the blur view is rendered multiple times, as mentioned here https://github.com/Kureev/react-native-blur/issues/396.

The 2nd library with a blur API is `@shopify/react-native-skia` but requires RN >=0.66 and we're on 0.64.4.

Possible solutions
- update RN to 0.66 and check if `@shopify/react-native-skia` works
- use some service to convert SVG to png
- use image placeholder if it's SVG and needs to be blurred

[YOMO-489]: https://emurgo.atlassian.net/browse/YOMO-489?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,6 +78,7 @@
   },
   "resolutions": {
     "react-devtools-core": "4.14.0",
+    "react-native-svg": "13.8.0",
```
Comment (michalszorad) [inline package.json]: [`react-native-blockies-svg`](https://github.com/Emurgo/react-native-blockies-svg) uses v7. Ideally `react-native-blockies-svg` would use `react-native-svg` as a peer dependency. @stackchain should i make a PR for [`react-native-blockies-svg`](https://github.com/Emurgo/react-native-blockies-svg) ?

**Comment by wolverineks** [comment]
have you looked into expo blurview?

**Comment by michalszorad** [comment]
@wolverineks I've looked into their documentation, and they support iOS only, so I've ruled that one out. Unless we want to blur the SVG on iOS and show a placeholder on Android?

**Comment by wolverineks** [comment]
> @wolverineks I've looked into their documentation, and they support iOS only, so I've ruled that one out. Unless we want to blur the SVG on iOS and show a placeholder on Android?

yeah, use a placeholder and let @stackchain know.

**Comment by stackchain** [comment]
> > @wolverineks I've looked into their documentation, and they support iOS only, so I've ruled that one out. Unless we want to blur the SVG on iOS and show a placeholder on Android?
> 
> yeah, use a placeholder and let @stackchain know.

@MichalSzorad, this is not a blocker for this PR, since moderation is behind a feature flag, but we definitely shouldn't use the same placeholder, it should be a different one that indicates that we have the image but we can't display it, please open a ticket for Product, so they can talk/sync with Design.

File: package.json (hunk 1 above)
Comment (stackchain) [reply package.json]: bumped it there, and re open here.

**Comment by stackchain** [comment]
~~Isn't an update in the Podfile lock file?~~

File: src/NftDetails/NftDetails.tsx [hunk 2]
```diff
@@ -61,21 +61,9 @@ export const NftDetails = () => {
...
-  disabled,
-}: {
-  source: ImageSourcePropType
-  onPress?: () => void
-  disabled?: boolean
-}) => {
   return (
-    <TouchableOpacity onPress={onPress} disabled={disabled} style={styles.imageWrapper}>
-      <Image source={source} style={styles.image} resizeMode="contain" />
+    <TouchableOpacity onPress={() => navigateTo.nftZoom(nft.id)} disabled={false} style={styles.imageWrapper}>
```
Comment (stackchain) [inline src/NftDetails/NftDetails.tsx]:
```suggestion
    <TouchableOpacity onPress={() => navigateTo.nftZoom(nft.id)} style={styles.imageWrapper}>
```

File: src/NftDetails/NftDetails.tsx [hunk 3]
```diff
@@ -86,12 +74,18 @@ const ModeratedNftImage = ({nft}: {nft: YoroiNft}) => {
   const {moderationStatus} = useModeratedNftImage({wallet, fingerprint: nft.fingerprint})
   const canShowNft = moderationStatus === 'approved' || moderationStatus === 'consent'
 
+  if (!canShowNft) {
```
Comment (stackchain) [inline src/NftDetails/NftDetails.tsx:77]: shouldn't it consider the moderation feature flag?

File: package.json (hunk 1 above)
Comment (banklesss) [reply package.json]: That will be good for the expo migration. 

Issues with react-native-svg versions

File: src/components/NftPreview/NftPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  resizeMode?: ImageResizeMode
+  blurRadius?: number
+}) => {
+  if (showPlaceholder) {
+    return <Image source={placeholder} style={[style, {width, height}]} resizeMode={resizeMode ?? 'contain'} />
+  }
+
+  const uri = showThumbnail ? nft.thumbnail : nft.image
+  const isUriSvg = uri.toLowerCase().endsWith('.svg') || isSvgMediaType(nft.metadata.originalMetadata.mediaType)
+  nft.metadata.originalMetadata?.files?.some((file) => file.src === uri && isSvgMediaType(file.mediaType))
```
Comment (banklesss) [inline src/components/NftPreview/NftPreview.tsx]: What it is being validated in this line?

Comment (banklesss) [reply src/components/NftPreview/NftPreview.tsx]: I think an operator is missing right?

File: src/components/NftPreview/NftPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  width?: number
+  resizeMode?: ImageResizeMode
+  blurRadius?: number
+}) => {
+  if (showPlaceholder) {
+    return <Image source={placeholder} style={[style, {width, height}]} resizeMode={resizeMode ?? 'contain'} />
+  }
+
+  const uri = showThumbnail ? nft.thumbnail : nft.image
+  const isUriSvg = uri.toLowerCase().endsWith('.svg') || isSvgMediaType(nft.metadata.originalMetadata.mediaType)
```
Comment (banklesss) [inline src/components/NftPreview/NftPreview.tsx:28]: In the line bellow an optional chaining operator is being used for `originalMetadata`

File: src/NftDetails/NftDetails.tsx (hunk 3 above)
Comment (michalszorad) [reply src/NftDetails/NftDetails.tsx:77]: The feature flag check is in the parent component, which decides to render between ModeratedNftImage and UnModeratedNftImage.

File: src/components/NftPreview/NftPreview.tsx (hunk 4 above)
Comment (michalszorad) [reply src/components/NftPreview/NftPreview.tsx]: It is to check if the file is an SVG

File: src/components/NftPreview/NftPreview.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+storiesOf('NftPreview', module)
+  .add('Default', () => <NftPreview nft={nft} width={200} height={200} />)
+  .add('Showing a placeholder', () => <NftPreview nft={nft} showPlaceholder width={200} height={200} />)
+  .add('Blurred image except SVG', () => <NftPreview nft={nft} width={200} height={200} blurRadius={20} />)
+  .add('Showing an SVG', () => (
+    <NftPreview nft={{...nft, image: 'https://www.svgrepo.com/show/501883/ice-cream.svg'}} width={200} height={200} />
+  ))
+  .add('Showing NFT thumbnail', () => (
+    <NftPreview
+      nft={{...nft, thumbnail: 'https://www.svgrepo.com/show/501883/ice-cream.svg'}}
```
Comment (stackchain) [inline src/components/NftPreview/NftPreview.stories.tsx:24]: next iteration lets have these images in the test/ or storybook folder, so stories don't hit the network

**Review by stackchain — APPROVED** [review]
lgtm

File: src/components/NftPreview/NftPreview.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  nft.metadata.originalMetadata.files?.some((file) => file.src === uri && isSvgMediaType(file.mediaType))
+
+  if (isUriSvg) {
+    // passing width or height with value undefined has a different behavior than not passing it at all
+    return (
+      <SvgUri
+        {...(width !== undefined ? {width} : undefined)}
+        {...(height !== undefined ? {height} : undefined)}
```
Comment (stackchain) [inline src/components/NftPreview/NftPreview.tsx]: not big fan, yet, are optional?
```suggestion
        {...(width && {width})}
```

Comment (stackchain) [reply src/components/NftPreview/NftPreview.tsx]: height is not optional

**Review by stackchain — COMMENTED** [review]
:arrow_up:

File: src/components/NftPreview/NftPreview.tsx (hunk 7 above)
Comment (stackchain) [reply src/components/NftPreview/NftPreview.tsx]: ` preserveAspectRatio="xMinYMin meet"` ?

File: package.json (hunk 1 above)
Comment (michalszorad) [reply package.json]: Here's the PR https://github.com/Emurgo/react-native-blockies-svg/pull/1

File: src/components/NftPreview/NftPreview.tsx (hunk 7 above)
Comment (michalszorad) [reply src/components/NftPreview/NftPreview.tsx]: @stackchain `xMinYMin meet` seems to work best https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio#example

**Review by stackchain — APPROVED** [review]

File: .tool-versions [hunk 8]
```diff
@@ -1,4 +1,4 @@
-nodejs 16.5.0
+nodejs 16.19.0
```
Comment (stackchain) [inline .tool-versions:1]: blake2 doesn't build on aarch64 on 16.5

File: package.json (hunk 1 above)
Comment (stackchain) [reply package.json]: merged and published

**Comment by stackchain** [comment]
@MichalSzorad I deployed the new version of blockies, yet, I think it's missing to add the same logic to the asset list to handle the svgs, can you check it?

File: .tool-versions (hunk 8 above)
Comment (michalszorad) [reply .tool-versions:1]: Thanks I see you've already updated the version

**Comment by michalszorad** [comment]
@stackchain the asset list is now fixed and I've also removed the old react-native-blockies-svg library

File: src/NftDetails/NftDetails.tsx [hunk 9]
```diff
@@ -86,12 +74,18 @@ const ModeratedNftImage = ({nft}: {nft: YoroiNft}) => {
...
+    return (
+      <View style={styles.imageWrapper}>
+        <NftPreview nft={nft} showPlaceholder={true} style={styles.image} height={380} />
```
Comment (banklesss) [inline src/NftDetails/NftDetails.tsx]:
```suggestion
        <NftPreview nft={nft} style={styles.image} height={380} showPlaceholder />
```

**Review by banklesss — APPROVED** [review]

File: src/components/NftPreview/NftPreview.tsx (hunk 5 above)
Comment (banklesss) [reply src/components/NftPreview/NftPreview.tsx:28]: https://github.com/Emurgo/yoroi-mobile/pull/2401/files#r1143508057

**Pull request merged**